WPThemeReview/Sniffs/ThouShallNotUse/ReservedTemplateNameSniff.php

```diff
@@ -0,0 +1,137 @@
+<?php
+/**
+ * WPThemeReview Coding Standard.
+ *
+ * @package WPTRT\WPThemeReview
+ * @link    https://github.com/WPTRT/WPThemeReview
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+namespace WPThemeReview\Sniffs\ThouShallNotUse;
+
+use PHP_CodeSniffer\Sniffs\Sniff;
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Util\Tokens;
+
+/**
+ * Check if the template file is using reserved file name as a part of their name.
+ *
+ * @since 0.2.0
+ */
+class ReservedTemplateNameSniff implements Sniff {
+
+	/**
+	 * Error message template.
+	 *
+	 * @var string
+	 */
+	const ERROR_MSG = 'File template should not use a reserved name as a prefix. Found: "%s".';
+
+	/**
+	 * Regex to catch the template creation comment
+	 *
+	 * @var string
+	 */
+	const TEMPLATE_REGEX = '`\/\* Template Name:`i';
+
+	/**
+	 * Found prefix in a file
+	 *
+	 * @var string
+	 */
+	protected $prefix;
+
+	/**
+	 * List of reserved template file names.
+	 *
+	 * @link https://developer.wordpress.org/themes/basics/template-hierarchy/
+	 * @link https://developer.wordpress.org/themes/template-files-section/partial-and-miscellaneous-template-files/#content-slug-php
+	 * @link https://wphierarchy.com/
+	 * @link https://en.wikipedia.org/wiki/Media_type#Naming
+	 *
+	 * @since 0.2.0
+	 *
+	 * @var array
+	 */
+	protected $reserved_file_name_prefixes = [ 'index-', 'comments-', 'front-page-', 'home-', 'header-', 'singular-', 'single-', 'page-', 'category-', 'tag-', 'taxonomy-', 'author-', 'date-', 'archive-', 'search-', 'attachment-', 'image-', '404-' ];
+
+	/**
+	 * Strip quotes surrounding an arbitrary string.
+	 *
+	 * Intended for use with the contents of a T_CONSTANT_ENCAPSED_STRING / T_DOUBLE_QUOTED_STRING.
+	 *
+	 * Used from WordPressCS\WordPress\Sniff abstract class-
+	 *
+	 * @since 0.2.0
+	 *
+	 * @param string $string The raw string.
+	 * @return string String without quotes around it.
+	 */
+	private function strip_quotes( $string ) {
+		return preg_replace( '`^([\'"])(.*)\1$`Ds', '$2', $string );
+	}
+
+	/**
+	 * Returns an array of tokens this test wants to listen for.
+	 *
+	 * @return array
+	 */
+	public function register() {
+		return Tokens::$emptyTokens;
+	}
+
+	/**
+	 * Processes this test, when one of its tokens is encountered.
+	 *
+	 * @param \PHP_CodeSniffer\Files\File $phpcsFile The PHP_CodeSniffer file where the
+	 *                                               token was found.
+	 * @param int                         $stackPtr  The position of the current token
+	 *                                               in the stack.
+	 *
+	 * @return void
```

Comment:
```suggestion
	 * @return int|void ....
```

See https://github.com/squizlabs/PHP_CodeSniffer/blob/1d7c6576a578fb57723c8c0fbc716cb88dae4f13/src/Sniffs/Sniff.php#L72-L75 for inspiration for the return type description.